packages/webapp/src/components/Form/NumberInput/utils.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+export function countDecimalPlaces(number: number) {
```

Comment:
Is this the best way to count decimal places? I was just trying to figure out what this does and happened to find that the function returns `3` for the input `1.11`.

Reply:
Oh I see what's happening there. Thank you for catching this! I will try to explain what its doing and what I did wrong.

This function is taking the number passed in and multiplying by e (power of 10) on each iteration and moving the decimal to the right by one place until it reaches the end. Using the % (modulo) operator, it checks if there is a remainder and stops when (number*e % 1) = 0, indicating that there are no more decimal places.

For example, if you pass in 1.11:

**First iternation:**
e=1
decimalPlaces=0
 (1.11 * 1) % 1 = 0.11

**Second iternation:**
e=10
decimalPlaces=1
 (1.11 * 10) % 1 = 0.1

**Third iternation:**
e=100
decimalPlaces=2
 (1.11 * 100) % 1 = 0

The thing I overlooked is that in JavaScript, sometimes when you multiply a decimal number it doesn't give the result with exact precision. Like 1.11*100 = 111.00000000001, not 111, so in the example above, third iteration is not true and the loop continues.

Reply:
I fixed it by rounding to a whole number (Math.Round(number*e)) and dividing by e, then checking if it equals the original number. Also added test cases for it and also for the clamp function.